SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialGameInstance.cpp

```diff
@@ -195,3 +203,55 @@ void USpatialGameInstance::HandleOnConnectionFailed(const FString& Reason)
 #endif
 	OnConnectionFailed.Broadcast(Reason);
 }
+
+void USpatialGameInstance::OnLevelInitializedNetworkActors(ULevel* LoadedLevel, UWorld* OwningWorld)
+{
+	const FString WorkerType = GetSpatialWorkerType().ToString();
+
+	if (OwningWorld != GetWorld()
+		|| !OwningWorld->IsServer()
+		|| !GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking()
+		|| (OwningWorld->WorldType != EWorldType::PIE
+			&& OwningWorld->WorldType != EWorldType::Game
+			&& OwningWorld->WorldType != EWorldType::GamePreview))
+	{
+		// We only want to do something if this is the correct process and we are on a spatial server, and we are in-game
+		return;
+	}
+
+	for (int32 ActorIndex = 0; ActorIndex < LoadedLevel->Actors.Num(); ActorIndex++)
+	{
+		AActor* Actor = LoadedLevel->Actors[ActorIndex];
+		if (Actor == nullptr)
+		{
+			continue;
+		}
+
+		if (USpatialStatics::IsSpatialOffloadingEnabled())
+		{
+			if (!USpatialStatics::IsActorGroupOwnerForActor(Actor))
+			{
+				if (!Actor->bNetLoadOnNonAuthServer)
+				{
+					Actor->Destroy();
+				}
+				else
+				{
+					UE_LOG(LogSpatialGameInstance, Verbose, TEXT("WorkerType %s is not the actor group owner of startup actor %s, exchanging Roles"), *WorkerType, *GetPathNameSafe(Actor));
+					ENetRole Temp = Actor->Role;
+					Actor->Role = Actor->RemoteRole;
+					Actor->RemoteRole = Temp;
+				}
+			}
+		}
+		else if (GetDefault<USpatialGDKSettings>()->bEnableUnrealLoadBalancer)
+		{
+			if (Actor->GetIsReplicated())
+			{
+				// Always wait for authority to be delegated down from SpatialOS, if using zoning
+				Actor->Role = ROLE_SimulatedProxy;
```

Comment:
We're only doing this now if the load balancer is enabled - which wasn't the case previously. Become authoritative over all actors should get called in either instance, so possibly safe to keep behaviour consistent.